.github/workflows/ci.yml

```diff
@@ -923,6 +923,38 @@ jobs:
         with:
           job-id: migration-tests-${{ matrix.old-version }}-${{ matrix.database }}
 
+  test-framework:
+    name: Keycloak Test Framework
+    runs-on: ubuntu-latest
+    needs: build
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: integration-test-setup
+        name: Integration test setup
+        uses: ./.github/actions/integration-test-setup
+
+      - name: Run tests
+        run: ./mvnw test -f test-framework/pom.xml
+
+      - name: Upload JVM Heapdumps
+        if: always()
+        uses: ./.github/actions/upload-heapdumps
+
+      - uses: ./.github/actions/upload-flaky-tests
+        name: Upload flaky tests
+        env:
+          GH_TOKEN: ${{ github.token }}
+        with:
+          job-name: Keycloak Test Framework
+
+      - name: Surefire reports
+        if: always()
+        uses: ./.github/actions/archive-surefire-reports
+        with:
+          job-id: test-framework
+
```

Comment:
This can all be removed. We don't need or use heapdumps, these tests should not be flaky, so shouldn't upload flaky results (nor are they being executed with re-runs), and uploading surefire reports is just a waste as you can just review the log output.